scripts/us_census/enhanced_tmcf/process_mappings.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+'''Processing the mappings + enhanced TMCF for US Census.'''
+
+import json
+import os
+import requests
+import sys
+import time
+
+from absl import app
+from absl import flags
+from typing import Dict, List
+from zipfile import ZipFile
+
+import process_etmcf
+
+# Allows the following module imports to work when running as a script
+_SCRIPT_PATH = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(_SCRIPT_PATH, '../..'))
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('zip_download_path', 'data/zip_download',
+                    'Zip file download directory')
+flags.DEFINE_string('input_directory', 'data/input', 'Inputs directory')
+flags.DEFINE_string('output_directory', 'data/output', 'Inputs directory')
+flags.DEFINE_string('mappings_file', 'mappings.txt', 'Mappings File.')
+flags.DEFINE_string('input_etmcf_filename', 'input_etmcf',
+                    'Enhanced TMCF filename.')
+
+NUM_RETRIES = 10
+
+
+def _get_zip_download_url(url: str, json_params: Dict) -> str:
+    x = requests.post(url, json=json_params)
+    print(x.content)
+    return json.loads(x.text)['response']['url']
+
+
+def _download_zip(url: str, filepath: str) -> None:
+    response = requests.get(url, stream=True)
+    with open(filepath, "wb") as f:
+        for chunk in response.iter_content(chunk_size=512):
+            if chunk:  # filter out keep-alive new chunks
+                f.write(chunk)
+
+    # Read the file back and check if it is a valid zip file.
+    # If it is not a zipfile, it will raise an Exception.
+    ZipFile(filepath, 'r')
+
+
+def _read_write_files_from_zip(zip_filepath, unzipped_data_path,
+                               csv_filename) -> None:
+    unzip_files = False
+
+    with ZipFile(zip_filepath, 'r') as z:
+        csv_files = []
+        for zip_file in z.namelist():
+            if not os.path.exists(os.path.join(unzipped_data_path,
+                                               csv_filename)):
+                if "Data.csv" in zip_file:
```

Comment:
use a flag for this, instead of hardcoding "Data.csv"

Reply:
done